sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ExecutingSqlQueries_NetCore.cs

```diff
@@ -14,32 +14,58 @@ public void Foo(DbContext context, string query, params object[] parameters)
-            context.Database.ExecuteSqlCommand(query); // Noncompliant
+            context.Database.ExecuteSqlCommand(query); // Compliant, not concat or format
             context.Database.ExecuteSqlCommand("" + query); // Noncompliant
-            context.Database.ExecuteSqlCommand($"", parameters); // Noncompliant, the FormattableString is evaluated before and is not sanitized
-            context.Database.ExecuteSqlCommand(query, parameters); // Noncompliant
+            context.Database.ExecuteSqlCommand($"", parameters); // Compliant, not concat or format
+            context.Database.ExecuteSqlCommand(query, parameters); // Compliant, not concat or format
             context.Database.ExecuteSqlCommand("" + query, parameters); // Noncompliant
 
             context.Database.ExecuteSqlCommandAsync($""); // Compliant, FormattableString is sanitized
             context.Database.ExecuteSqlCommandAsync(""); // Compliant, constants are safe
             context.Database.ExecuteSqlCommandAsync(ConstQuery); // Compliant, constants are safe
             context.Database.ExecuteSqlCommandAsync("" + ""); // Compliant, constants are safe
-            context.Database.ExecuteSqlCommandAsync(query); // Noncompliant
+            context.Database.ExecuteSqlCommandAsync(query); // Complinat, not concat
             context.Database.ExecuteSqlCommandAsync("" + query); // Noncompliant
-            context.Database.ExecuteSqlCommandAsync($"", parameters); // Noncompliant, the FormattableString is evaluated before and is not sanitized
-            context.Database.ExecuteSqlCommandAsync(query, parameters); // Noncompliant
+            context.Database.ExecuteSqlCommandAsync($"", parameters); // Compliant, not concat or format
+            context.Database.ExecuteSqlCommandAsync(query, parameters); // Compliant, not concat or format
             context.Database.ExecuteSqlCommandAsync("" + query, parameters); // Noncompliant
 
             context.Query<User>().FromSql($""); // Compliant, FormattableString is sanitized
             context.Query<User>().FromSql(""); // Compliant, constants are safe
             context.Query<User>().FromSql(ConstQuery); // Compliant, constants are safe
-            context.Query<User>().FromSql(query); // Noncompliant
+            context.Query<User>().FromSql(query); // Compliant, not concat/format
             context.Query<User>().FromSql("" + ""); // Compliant
-            context.Query<User>().FromSql($"", parameters); // Noncompliant, the FormattableString is evaluated before and is not sanitized
+            context.Query<User>().FromSql($"", parameters); // Compliant
             context.Query<User>().FromSql("", parameters); // Compliant, the parameters are sanitized
-            context.Query<User>().FromSql(query, parameters); // Noncompliant, even though the parameters are sanitized, query could be tainted
+            context.Query<User>().FromSql(query, parameters); // Compliant
```

Comment:
I can't see a test for one of the sensitive examples given in the RSPEC:

`context.Database.ExecuteSqlCommand($"SELECT * FROM mytable WHERE mycol={value}", param); // Sensitive, the FormattableString is evaluated and converted to RawSqlString`

Reply:
good catch!